sdk/storage/azure-storage-blob-batch/src/main/java/com/azure/storage/blob/batch/BlobBatchAsyncClient.java

```diff
@@ -0,0 +1,113 @@
+            .build();
+    }
+
+    /**
+     * Submits a batch operation.
+     *
+     * <p>If any request in a batch fails this will throw a {@link StorageException}.</p>
+     *
+     * <p><strong>Code samples</strong></p>
+     *
+     * {@codesnippet com.azure.storage.blob.BlobServiceAsyncClient.submitBatch#BlobBatch}
+     *
+     * @param batch Batch to submit.
+     * @return An empty response indicating that the batch operation has completed.
+     * @throws StorageException If any request in the {@link BlobBatch} failed or the batch request is malformed.
+     */
+    public Mono<Void> submitBatch(BlobBatch batch) {
+        return withContext(context -> submitBatchWithResponse(batch, true, context)).flatMap(FluxUtil::toMono);
+    }
+
+    /**
+     * Submits a batch operation.
+     *
+     * <p>If {@code throwOnAnyFailure} is {@code true} a {@link StorageException} will be thrown if any request
+     * fails.</p>
+     *
+     * <p><strong>Code samples</strong></p>
+     *
+     * {@codesnippet com.azure.storage.blob.BlobServiceAsyncClient.submitBatchWithResponse#BlobBatch-boolean}
+     *
+     * @param batch Batch to submit.
+     * @param throwOnAnyFailure Flag to indicate if an exception should be thrown if any request in the batch fails.
+     * @return A response only containing header and status code information, used to indicate that the batch operation
+     * has completed.
+     * @throws StorageException If {@code throwOnAnyFailure} is {@code true} and any request in the {@link BlobBatch}
+     * failed or the batch request is malformed.
+     */
+    public Mono<Response<Void>> submitBatchWithResponse(BlobBatch batch, boolean throwOnAnyFailure) {
+        return withContext(context -> submitBatchWithResponse(batch, throwOnAnyFailure, context));
+    }
+
+    Mono<Response<Void>> submitBatchWithResponse(BlobBatch batch, boolean throwOnAnyFailure, Context context) {
+        return postProcessResponse(this.client.services().submitBatchWithRestResponseAsync(
+            batch.getBody(), batch.getContentLength(), batch.getContentType(), context))
+            .flatMap(response -> BlobBatchHelper.mapBatchResponse(batch, response, throwOnAnyFailure));
+    }
+
+    /**
+     * Deletes the passed blobs.
+     *
+     * @param blobUrls Blobs to delete.
+     * @return
+     * @throws StorageException If the batch is malformed or any of the delete operations fail.
+     */
+    public Flux<Response<Void>> deleteBlobs(String... blobUrls) {
+        return batchingHelper(BlobBatch::delete, blobUrls);
+    }
+
+    /**
+     * Sets the {@link AccessTier} on the passed blobs.
+     *
+     * @param accessTier {@link AccessTier} to set on the blobs.
+     * @param blobUrls Blobs to set the access tier.
+     * @return
+     * @throws StorageException If the batch is malformed or any of the set access tier operations fail.
+     */
+    public Flux<Response<Void>> setBlobsAccessTier(AccessTier accessTier, String... blobUrls) {
```

Comment:
>(AccessTier accessTie [](start = 50, length = 21)

Possible alternative is accepting a list of tiers and a parallel list of blobUrls to make it possible for each blob to get a specific tier. I don't feel strongly about this though.

Reply:
For now I am going to move forward with the single access tier and list of blob urls to stay aligned with .NET, in the future if feedback requests that functionality it'll be an easy add.